pkg/apis/kops/validation/legacy.go

```diff
@@ -537,17 +537,11 @@ func DeepValidate(c *kops.Cluster, groups []*kops.InstanceGroup, strict bool) er
 	}
 
 	for _, g := range groups {
-		errs := CrossValidateInstanceGroup(g, c)
+		errs := CrossValidateInstanceGroup(g, c, cloud)
 
-		// Additional cloud-specific validation rules,
-		// such as making sure that identifiers match the expected formats for the given cloud
-		switch kops.CloudProviderID(c.Spec.CloudProvider) {
-		case kops.CloudProviderAWS:
-			errs = append(errs, awsValidateInstanceGroup(g)...)
```

Comment:
`awsValidateInstanceGroup` is now called through `CrossValidateInstanceGroup` -> `ValidateInstanceGroup` so that we can get additional validation for `kops create ig` and `kops edit ig` since they only call `CrossValidateInstanceGroup` rather than `DeepValidate`.